### PR TITLE
feat(event-tags): Add highlight fields to settings page

### PR DESCRIPTION
**Pull request by leeandher**
~~Requires https://github.com/getsentry/sentry/pull/69044~~
~~Requires https://github.com/getsentry/sentry/pull/69426~~

Adds some fields to the project settings page to modify tags/context highlights directly. 

**Update:** It's been moved to the Tags page, and that page title was changed to "Tags & Context"

image

Without the flag, just the sidebar title and heading are still changed.

image

**todo**
- [x] Add tests 
- [x] Ensure the form displays with current settings and updates on blur

### CODE REVIEW

File: static/app/views/settings/project/navigationConfiguration.tsx [hunk 1]
```diff
@@ -39,8 +39,8 @@ export default function getConfiguration({
         },
         {
           path: `${pathPrefix}/tags/`,
-          title: t('Tags'),
-          description: t("View and manage a  project's tags"),
+          title: t('Tags & Context'),
+          description: t("View and manage a project's tags and highlights"),
```
Comment (malwilley) [inline static/app/views/settings/project/navigationConfiguration.tsx:43]: Is it okay to mention this even without the feature flag enabled?

File: static/app/components/events/highlights/highlightsSettingsForm.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      method: 'PUT',
+      body: {...project, highlightTags, highlightContext: newContext},
+    });
+
+    expect(screen.getByText('Highlighted Context')).toBeInTheDocument();
+    const contextInput = screen.getByRole('textbox', {name: 'Highlighted Context'});
+
+    // React Testing Library requires these characters to be escaped by having two
+    const rtlEscapeCharacters = new Set(['{', '[']);
+    const rtlSafeNewContext = JSON.stringify(newContext)
```
Comment (malwilley) [inline static/app/components/events/highlights/highlightsSettingsForm.spec.tsx:72]: If you wanted to skip this escapting, I think you could just do `userEvent.paste()`. I don't think typing every character provides any additional value for the test

Comment (leeandher) [reply static/app/components/events/highlights/highlightsSettingsForm.spec.tsx:72]: TIL, thanks!

File: static/app/views/settings/project/navigationConfiguration.tsx (hunk 1 above)
Comment (leeandher) [reply static/app/views/settings/project/navigationConfiguration.tsx]: good call, replaced with `and context`, thanks!

**Review by malwilley — APPROVED** [review]

**Pull request merged**